### DISCUSSION TITLE
Bicep dependencies should be considered case-insensitive.

### DISCUSSION THREAD

**Discussion by trisjedwards** [renovatebot/renovate#21188] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

![image](https://user-images.githubusercontent.com/963489/227913791-40516524-b94f-4c27-9b21-32fad1a39829.png)
This is from an update to our bicep file, that contains two resources from `Microsoft.Insights/scheduledQueryRules`, but they were added by different people using different casings.

I don't think Renovate should consider them separate updates in this case.

**Comment by rarkins** [comment]
If they are in separate files, then they _are_ separate updates. So your proposed change is cosmetic, i.e. duplicate them in the table from your screenshot?

**Reply by trisjedwards** [reply]
Sorry, I was not totally clear. These are in the _same_ file, but listed as different updates _only_ because the casing is different.
In other cases in bicep where multiple resources are updated for the same version, only one is shown in the table as you would expect.
This is not just a cosmetic change, as package rules currently only apply to the exact casing match. 

I believe desired behaviour should be: 

`"matchPackageNames": ['Microsoft.Insights/scheduledQueryRules']` should match any casing of this package, since they are all valid and point to the same package.

**Reply by rarkins** [reply]
Asking for package rules to be case insensitive is a totally different topic. The bicep manager can't control that, other than it could perhaps lower case _all_ dependencies it extracts

**Reply by trisjedwards** [reply]
After peeking at the package rules code, I see how that would be very difficult. 

In that case just the grouping of updates in the PR to reduce noise would be beneficial, and as a workaround for package rules I can use `matchPackagePatterns` to make a case-insensitive regex expression. 

Would something like `"matchPackagePatterns": ["/Microsoft.Insights\\/scheduledQueryRules/i"]` work in this case? I'm not sure how the  regex grouping interacts with escaped characters.

**Reply by rarkins** [reply]
I'm not sure if we added support for modifiers like `i`, but if not then we could hopefully add a special case at least for ending in `/i`

**Reply by viceice** [reply]
we don't support modifiers

**Reply by rarkins** [reply]
I'd support a feature request for this
